pmd-java/etc/grammar/Java.jjt

```diff
@@ -2415,20 +2470,47 @@ void StatementExpression() :
   PostfixExpression()
 }
 
-void SwitchStatement() :
+void SwitchStatement():
+{}
+{
+  "switch" "(" Expression() ")" SwitchBlock()
+}
+
+void SwitchBlock() #void :
+{}
+{
+    "{"
+        (
+            LOOKAHEAD( { switchLabeledRuleLookahead(10) } ) ( SwitchLabeledRule() )+
```

Comment:
This lookahead means that `case A,B,C,D,E,F,G->` doesn't parse...

Reply:
Yes, that's a limitation... it would not only limit the number of "simple" expressions, but there could more complex constant expressions, which take more than 1 token each...

I think, your solution is smarter :)